### PR TITLE
EB-18050: Add checks for doctrine classes

### PR DESCRIPTION
**Pull request by graillus**
- [x] Add check to forbid usage of Doctrine classes in domain code such as `EntityManager`, `DocumentManager`, `ManagerRegistry`, ...
- [x] Add check to forbid direct injection of doctrine repositories in constructor/properties

### CODE REVIEW

**Comment by graillus** [comment]
Example errors in SESSION:

image

File: src/ETSGlobalCodingStandard/Sniffs/Doctrine/ForbidRepositoryInjectionSniff.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    public const FORBIDDEN_REPOSITORY_INJECTION = 'ForbiddenRepositoryInjection';
+    public const FORBIDDEN_REPOSITORY_PROPERTY = 'ForbiddenRepositoryProperty';
+
+    /**
+     * All classes within this namespace will be considered to be Doctrine repositories.
+     *
+     * @var string[]
+     */
+    public array $repositoryNamespaces = [
+        'App\Repository',
```
Comment (ch3ric) [inline src/ETSGlobalCodingStandard/Sniffs/Doctrine/ForbidRepositoryInjectionSniff.php:35]: this property will be editable from our apps CS xml configs in order to have different values depending on the app?

File: src/ETSGlobalCodingStandard/Sniffs/Doctrine/ForbidDoctrineClassesSniff.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    ];
+
+    /**
+     * List of namespaces allowed to use the forbidden classes.
+     * Add a trailing '\' to also ignore all sub-namespaces within it.
+     *
+     * @var string[]
+     */
+    public array $ignoredNamespaces = [
+        'App\Migration',
```
Comment (ch3ric) [inline src/ETSGlobalCodingStandard/Sniffs/Doctrine/ForbidDoctrineClassesSniff.php:41]: this property will be editable from our apps CS xml configs in order to have different values depending on the app?

File: src/ETSGlobalCodingStandard/Sniffs/Doctrine/ForbidRepositoryInjectionSniff.php (hunk 1 above)
Comment (graillus) [reply src/ETSGlobalCodingStandard/Sniffs/Doctrine/ForbidRepositoryInjectionSniff.php:35]: Yes, this is what the config should look line in SESSION:
```xml
    <rule ref="ETSGlobalCodingStandard.Doctrine.ForbidDoctrineClasses">
        <properties>
            <property name="ignoredNamespaces" type="array">
                <element value="AppBundle\DataFixtures"/>
                <element value="ETSGlobal\Infrastructure\Repository"/>
                <element value="Tests\"/>
            </property>
        </properties>
    </rule>
    <rule ref="ETSGlobalCodingStandard.Doctrine.ForbidRepositoryInjection">
        <properties>
            <property name="repositoryNamespaces" type="array">
                <element value="ETSGlobal\Domain\Repository"/>
                <element value="ETSGlobal\Infrastructure\Repository"/>
            </property>
        </properties>
    </rule>
```

**Review by ch3ric — DISMISSED** [review]

**Review by lepiaf — APPROVED** [review]

**Review by aallier — APPROVED** [review]

**Comment by ch3ric** [comment]
:heavy_check_mark:

**Pull request merged**